packages/rich-list-core/src/lib/PersistentLinkedList.ts

```diff
@@ -0,0 +1,29 @@
+export interface PersistentLinkedList<T> {
+  append: (data: T) => Promise<void>
+  removeLast: () => Promise<void>
+  getLast: () => Promise<T | undefined>
+  size: () => Promise<number>
+}
+
+/**
+ * For unit test use.
+ */
+export class InMemoryLinkedListService<T> implements PersistentLinkedList<T> {
+  private readonly DATA: T[] = []
+
+  async append (data: T): Promise<void> {
+    this.DATA.push(data)
+  }
+
+  async removeLast (): Promise<void> {
+    this.DATA.pop()
+  }
+
+  async getLast (): Promise<T | undefined> {
+    return this.DATA[this.DATA.length - 1]
+  }
+
+  async size (): Promise<number> {
+    return this.DATA.length
+  }
+}
```

Comment:
```suggestion
export class InMemoryLinkedListService<T> implements PersistentLinkedList<T> {
  private readonly data: T[] = []

  async append (data: T): Promise<void> {
    this.data.push(data)
  }

  async removeLast (): Promise<void> {
    this.data.pop()
  }

  async getLast (): Promise<T | undefined> {
    return this.data[this.data.length - 1]
  }

  async size (): Promise<number> {
    return this.data.length
  }
}
```
Based on previous insights I think this can be renamed to lowercased `data`

Reply:
It is meant to be readonly, since no re-write needed. And this is a stubbed service for test use, I guess it is fine.